pkg/sql/conn_executor_prepare.go

```diff
@@ -53,11 +53,20 @@ func (ex *connExecutor) execPrepare(
 	// type OIDs into types.T's.
 	if parseCmd.TypeHints != nil {
 		for i := range parseCmd.TypeHints {
-			if parseCmd.TypeHints[i] == nil && types.IsOIDUserDefinedType(parseCmd.RawTypeHints[i]) {
-				var err error
-				parseCmd.TypeHints[i], err = ex.planner.ResolveTypeByOID(ctx, parseCmd.RawTypeHints[i])
-				if err != nil {
-					return retErr(err)
+			if parseCmd.TypeHints[i] == nil {
+				if i >= len(parseCmd.RawTypeHints) {
+					return retErr(pgerror.Newf(
+						pgcode.ProtocolViolation,
+						"expected type hint at position %d",
```

Comment:
maybe we should try to match the error message that happens in the simple case:

https://github.com/cockroachdb/cockroach/blob/32bc171667b6e180fa09669802d45fc9edbe4a52/pkg/sql/conn_executor_prepare.go#L300